sdk/inc/azure/core/az_json.h

```diff
@@ -88,7 +88,9 @@ AZ_NODISCARD az_result az_json_token_get_boolean(az_json_token const* json_token
  * @param json_token A pointer to an #az_json_token instance.
  * @param out_value A pointer to a variable to receive the value.
  * @return AZ_OK if the number is returned.<br>
- * AZ_ERROR_JSON_INVALID_STATE if the kind != AZ_JSON_TOKEN_NUMBER.
+ * AZ_ERROR_JSON_INVALID_STATE if the kind != AZ_JSON_TOKEN_NUMBER.<br>
+ * AZ_ERROR_PARSER_UNEXPECTED_CHAR if a non-ASCII digit is found within the token or if it contains
+ * a number that would overflow or underflow uint64
```

Comment:
The JSON stacks within other sdks, like .NE, do not distinguish overflow from unexpected char.

For example, all of these throw a format exception when trying to turn into an int32:
- 123.1
- 123A
- 99999999999

Reply:
We could decide to split the error condition and add a new `az_result`, if you think that would be better, but detecting that cleanly, especially for `double` is complicated and would potentially bloat the code size (since we are primarily relying on sscanf to do the heavy lifting here).